pkg/operator/mirror.go

```diff
@@ -430,103 +445,67 @@ func (o *MirrorOptions) associateDeclarativeConfigImageLayers(ctlgRef imagesourc
 		return nil
 	}
 
-	foundAtLeastOneMapping := false
-	if err := filepath.Walk(o.tmp, func(path string, info fs.FileInfo, err error) error {
-		if err != nil || info.IsDir() {
-			return err
-		}
-		if !isMappingFile(path) {
-			return nil
-		}
-		foundAtLeastOneMapping = true
-
-		o.Logger.Debugf("reading mapping file %s", path)
-		imgMappings, err := image.ReadImageMapping(path)
-		if err != nil {
-			return err
-		}
-
-		for _, err := range []error{
-			// NB(estroz): assuming that any related image will not point to
-			// an operator catalog or bundle may not be a valid assumption.
-			// Even if not valid, mirroring bundle and catalog images as normal images
-			// should work as though they were mirrored via the `catalog mirror` API.
-			associateWithType(imgMappings, relatedImages, image.TypeOperatorRelatedImage),
-			associateWithType(imgMappings, bundleImages, image.TypeOperatorBundle),
-			associateWithType(imgMappings, []string{ctlgRef.Ref.Exact()}, image.TypeOperatorCatalog),
```

Comment:
@estroz We maybe have to keep the `TypeOperatorCatalog` differentiation in here because that is how we are creating the catalog sources on the publishing side. That is why they (catalog sources) are not being created on this branch right now.

Reply:
I can fix this by generating CS's for each catalog image rebuilt.